scripts/ultimate-upscale.py

```diff
@@ -20,21 +20,23 @@ class USDUSFMode(Enum):
 
 class USDUpscaler():
 
-    def __init__(self, p, image, upscaler_index:int, save_redraw, save_seams_fix, tile_size) -> None:
+    def __init__(self, p, image, upscaler_index:int, save_redraw, save_seams_fix, tile_width, tile_height) -> None:
         self.p:StableDiffusionProcessing = p
         self.image:Image = image
         self.scale_factor = max(p.width, p.height) // max(image.width, image.height)
         self.upscaler = shared.sd_upscalers[upscaler_index]
         self.redraw = USDURedraw()
         self.redraw.save = save_redraw
-        self.redraw.tile_size = tile_size
+        self.redraw.tile_width = tile_width if tile_width > 0 else tile_height
+        self.redraw.tile_height = tile_height if tile_height > 0 else tile_width
         self.seams_fix = USDUSeamsFix()
         self.seams_fix.save = save_seams_fix
-        self.seams_fix.tile_size = tile_size
+        self.seams_fix.tile_width = tile_width if tile_width > 0 else tile_height
+        self.seams_fix.tile_height = tile_height if tile_height > 0 else tile_width
         self.initial_info = None
-        self.rows = math.ceil(self.p.height / tile_size)
-        self.cols = math.ceil(self.p.width / tile_size)
-        
+        self.rows = math.ceil(self.p.height / (tile_height if tile_height > 0 else tile_width))
```

Comment:
use self.redraw.tile_height instead of (tile_height if tile_height > 0 else tile_width)